Implement RUSTPYTHONPATH

OddBloke opened the pull request:
This mirrors PYTHONPATH for CPython.

cthulahoops commented:
Why RUSTPYTHONPATH rather than just PYTHONPATH? Won't that make us incompatible with things that try to manage the environment?

For reference, PyPy uses PYTHONPATH.

windelbouwman commented:
Agreed, we should go for PYTHONPATH

windelbouwman commented:
Fantastic change btw, with this we could try parts of the standard library!

windelbouwman commented:
Should PYTHONPATH be added to `sys.path`?

OddBloke commented:
My thinking was that PYTHONPATH could lead to annoying conflicts and that we could switch to it once we were closer to parity. That said, I was assuming that PYTHONPATH always had something in it, but that isn't true on my local system, so I'll switch to that.

And, yes, we probably should modify the import logic to use sys.path, and then PYTHONPATH handling can happen there; I'll also do that.

Thanks for the reviews!

OddBloke commented:
I believe I have addressed the comments; I've found one issue with our import logic that I've filed #80 for, because I think it might take a bit of effort to fix.
